embed/etcd.go

```diff
@@ -147,6 +161,7 @@ func (e *Etcd) Config() Config {
 
 func (e *Etcd) Close() {
 	e.closeOnce.Do(func() { close(e.stopc) })
+	e.OnShutdown()
```

Comment:
can the function be inlined here instead of needing a separate `OnShutdown` field?